### PR TITLE
Refactor gather/scatter benchmarks for strings

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Combines the `benchmarks/string/copy.cu` and `benchmarks/string/gather.cpp` source files which both had separate gather benchmarks for strings. The result is a new `copy.cpp` that has both gather and scatter benchmarks. Also changes the default parameters to remove the need to restrict the values.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/string/copy.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  auto const row_width = static_cast<cudf::size_type>(state.get_int64("row_width"));
+  auto const api       = state.get_string("api");
+
+  data_profile const table_profile = data_profile_builder().distribution(
+    cudf::type_id::STRING, distribution_id::NORMAL, 0, row_width);
+  auto const source =
+    create_random_table({cudf::type_id::STRING}, row_count{num_rows}, table_profile);
+
+  data_profile const map_profile = data_profile_builder().no_validity().distribution(
+    cudf::type_id::INT32, distribution_id::UNIFORM, 0, num_rows);
```
Comment (bdice) [inline cpp/benchmarks/string/copy.cpp:37]:
```suggestion
    cudf::type_to_id<cudf::size_type>(), distribution_id::UNIFORM, 0, num_rows);
```

**Review by bdice — APPROVED** [review]
Maybe `size_type` instead of `INT32`. If there are reasons to keep `INT32`, this LGTM.

File: cpp/benchmarks/string/copy.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
```
Comment (PointKernel) [inline cpp/benchmarks/string/copy.cpp:2]:
```suggestion
 * Copyright (c) 2024, NVIDIA CORPORATION.
```

**Review by PointKernel — APPROVED** [review]

**Review by lamarrr — APPROVED** [review]

**Pull request merged**